### PR TITLE
Add rules to ensure users are using pod_template_file for KubernetesE…

### LINKED ISSUE #11049: Create KubernetesWorkerAnnotationsRule to ease upgrade to Airflow 2.0
This issue is part of #8765

## Rule
Create `KubernetesWorkerAnnotationsRule` which corresponds to
> Changes to propagating Kubernetes worker annotations

entry in UPDATING.md. This rule should allow users to check if their current configuration needs any adjusting
before migration to Airflow 2.0.

## How to guide
To implement a new rule, create a class that inherits from `airflow.upgrade.rules.base_rule.BaseRule`.
It will be auto-registered and used by `airflow upgrade-check` command. The custom rule class has to have `title`,
`description` properties and should implement `check` method which returns a list of error messages in case of
incompatibility.

For example:
https://github.com/apache/airflow/blob/ea36166961ca35fc51ddc262ec245590c3e236fb/airflow/upgrade/rules/conn_type_is_not_nullable.py#L25-L42

**Remember** to open the PR against `v1-10-test` branch.

### PR DESCRIPTION
**Pull request by dimberman**
…xecutor

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by dimberman** [comment]
@turbaszek @mik-laj Is this the correct way to create rules? I can modify as needed then add more

**Comment by turbaszek** [comment]
Looks good to me, but will require #11081 to be fully functional

File: airflow/upgrade/rules/pod_template_file_rule.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+In Airflow 2.0, KubernetesExecutor Users need to set a pod_template_file as a base
+value for all pods launched by the KubernetesExecutor
+"""
+
+    def check(self):
+        pod_template_file = conf.get("kubernetes", "pod_template_file", fallback=None)
+        if not pod_template_file:
+            return (
+                "Please create a pod_template_file by running `airflow generate_pod_template`.\n"
+                "This will generate a pod using your aiflow.cfg settings"
```
Comment (turbaszek) [inline airflow/upgrade/rules/pod_template_file_rule.py:38]: Is there any link that we can use to point users to "how to use this template"?

**Comment by turbaszek** [comment]
Does this duplicate #11068 @dimberman @ephraimbuddy ?

**Comment by dimberman** [comment]
@turbaszek I don't think it does, as that PR is based on an update that is no longer a part of 2.0 (since we removed that config with Simplify K8sExecutor). That said, @ephraimbuddy, do I need to modify the UPDATING.md? Is it giving bad info?

**Comment by ephraimbuddy** [comment]
> @turbaszek I don't think it does, as that PR is based on an update that is no longer a part of 2.0 (since we removed that config with Simplify K8sExecutor). That said, @ephraimbuddy, do I need to modify the UPDATING.md? Is it giving bad info?

There's need to remove the section I'm working on about Kubernetes_annotations

**Comment by turbaszek** [comment]
@dimberman do I correctly understand that this note:
https://github.com/apache/airflow/blob/master/UPDATING.md#changes-to-propagating-kubernetes-worker-annotations

is no longer valid for Airflow 2.0 and should be removed?

**Comment by dimberman** [comment]
@turbaszek yes that is correct. I can remove it in a bit or you can do it.

**Comment by turbaszek** [comment]
@dimberman I think we can do this as part of this PR

**Comment by turbaszek** [comment]
Hey @dimberman why so many files? 😄

**Pull request merged**